docker/Gemfile

```diff
@@ -0,0 +1,13 @@
+source ENV['GEM_SOURCE'] || "https://rubygems.org"
+
+def location_for(place)
+  if place =~ /^(git[:@][^#]*)#(.*)/
+    [{ :git => $1, :branch => $2, :require => false }]
```

Comment:
Not sure exactly how this works nor whether the tool(s) really care, but, we have a change pending to the puppet entry lower in the file to use :ref instead of :branch because for the integration testing we want to be able to use hashes, tags, or branches.

Reply:
We shouldn't need that in this case I don't think, but easy enough to change if that comes up. This is mostly for use in local testing, where I'm usually running off a file or a git branch I've pushed to my fork.